src/sql/workbench/browser/modelComponents/tree.component.ts

```diff
@@ -162,6 +162,13 @@ export default class TreeComponent extends ComponentBase<azdata.TreeProperties>
 		if (this._treeRenderer) {
 			this._treeRenderer.options.withCheckbox = this.withCheckbox;
 		}
+
+		if (this._tree) {
+			// If tree was already initialized, update its properties
+			if (this.ariaLabel) {
+				this._tree.ariaLabel = this.ariaLabel;
```

Comment:
My worry here is - how does `treeComponent.ariaLabel = 'Foobar'` will end up updating the underlying div? I looked at other components and they all seem to be setting the `aria-label` of the underlying HTML element in this `setProperties` override, so I did the same here. I had to expose it from the tree control itself, as it owns the `div`.

It is worth mentioning that `TreeContext` also maintains the copy of options. We do not update that currently when `ariaLabel` is changing and it always represents the original options. I think this is reasonable, although I don't know what is the relationship between component, view and context here. Maybe context should maintain the current state of the object? Then it's a double-maintenance, as we will have actual HTML element that is the source of truth and has a pretty good chance to diverge from the context. I think just claiming that `context.options` will maintain the original snapshot of options is reasonable.

Reply:
Just having this be 

`this._tree?.getHTMLElement().setAttribute('aria-label', this.ariaLabel);`

is fine.

The initial context just being a snapshot of the starting values is fine - that's pretty standard from what I've seen where the options are just used to initialize the control and can be overwritten later as needed.

Reply:
Hm... My thinking was that TypeScript is not smart enough to generate common `if` block when you use null propagation multiple times, is it? People tend to copy-paste the code, continuous refactoring does not exist anymore, so we need to provide something that will be robust in the modern days software engineering... :) Technically we should do this for other `aria` properties too in here, it's just not required by this change, but I do see this coming some time in the future.